NEWS.md

```diff
@@ -439,6 +439,11 @@ Deprecated or removed
 
   * `cummin` and `cummax` have been deprecated in favor of `accumulate`.
 
+  * The idiosyncratic `Array` constructor syntax `Array(T, dims...)` has been deprecated
```

Comment:
the old syntax does match how you would call `zeros` or `ones` ... though those aren't constructors